c10/core/AllocatorConfig.h

```diff
@@ -310,6 +326,17 @@ class C10_API AcceleratorAllocatorConfig {
   // their own environment configuration extensions.
   inline static std::function<void(const std::string&)>
       device_config_parser_hook_{nullptr};
+
+  // A set of valid configuration keys, including both common and
+  // device-specific options. This set is used to validate the presence and
+  // correctness of keys during parsing.
+  inline static std::unordered_set<std::string> keys_{
+      "max_split_size_mb",
+      "max_non_split_rounding_mb",
+      "garbage_collection_threshold",
+      "roundup_power2_divisions",
+      "expandable_segments",
+      "pinned_use_background_threads"};
 };
```

Comment:
I've also been commenting on https://github.com/pytorch/pytorch/pull/150312, but this static initialization at class scope using a data structure from the STL is not safe in Windows DLLs. This could be moved into function scope or it could be initialized lazily using `std::once_flag` (`c10::once_flag`). This case is a bit trickier than the one in that other PR since `keys_` is mutated by `registerDeviceConfigParserHook()`

* https://stackoverflow.com/a/5115008
* https://isocpp.org/wiki/faq/ctors#static-init-order